### PR TITLE
Migrate `CompatibilityOptions` to normal ES6

### PR DESCRIPTION
**Pull request by james-pre**
This PR migrates `CompatibilityOptions` from a class with static properties into normal ES6 exports. This is from #14805.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

File: packages/dev/core/src/Compat/compatibilityOptions.ts [hunk 1]
```diff
@@ -1,9 +1,6 @@
+/* Options used to control default behaviors regarding compatibility support */
+
 /**
- * Options used to control default behaviors regarding compatibility support
+ * Defines if the system should use OpenGL convention for UVs when creating geometry or loading .babylon files (false by default)
  */
-export class CompatibilityOptions {
```
Comment (deltakosh) [inline packages/dev/core/src/Compat/compatibilityOptions.ts:4]: unfortunately this is a breaking change.

Comment (deltakosh) [reply packages/dev/core/src/Compat/compatibilityOptions.ts:4]: THAT BEING SAID, @sebavan this is purely internal right?

Comment (sebavan) [reply packages/dev/core/src/Compat/compatibilityOptions.ts:4]: It is exported so hard to tell if used publicly :-( It is a breaking change in its current form and wondering what does it bring pragmatically ?

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15376/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15376/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15376/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15376/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15376/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15376/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15376/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15376/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15376/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15376/merge/testResults/webgpuplaywright/index.html

File: packages/dev/core/src/Compat/compatibilityOptions.ts (hunk 1 above)
Comment (james-pre) [reply packages/dev/core/src/Compat/compatibilityOptions.ts:4]: This is more of a cleanup thing. If you like, I can keep the class but mark it as deprecated?

Comment (sebavan) [reply packages/dev/core/src/Compat/compatibilityOptions.ts:4]: Let s see what @RaananW thinks as well

Comment (james-pre) [reply packages/dev/core/src/Compat/compatibilityOptions.ts:4]: @deltakosh @sebavan,

I'm going to include an object for backward compatibility. It is marked as deprecated. (https://github.com/BabylonJS/Babylon.js/pull/15376/commits/57728a71a7dede24eccae72d220c440666b94e0e)

**Comment by deltakosh** [comment]
Wait:) why this Pr at the end of the day? What is the gain again?

**Review by sebavan — COMMENTED** [review]
There is still no back compat cause CompatibilityOptions now only use the default useOpenGLOrientationForUV not the one changed by the user.

**Comment by sebavan** [comment]
Let me close the PR as it is not worth the change. If the class with static sounds too much, you could simply replace with an object but keep the same structure.

**Comment by bjsplat** [comment]
Visualization tests for WebGL 1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15376/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by james-pre** [comment]
> There is still no back compat cause CompatibilityOptions now only use the default useOpenGLOrientationForUV not the one changed by the user.

Thanks for catching that, I missed it.

**Comment by james-pre** [comment]
> Wait:) why this Pr at the end of the day? What is the gain again?

I think #14805 explains the problem/solution quite well.

**Pull request closed without merging**